WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m

```diff
@@ -60,37 +64,72 @@ - (BOOL)fb_isVisible
   if (CGRectIsEmpty(frame)) {
     return NO;
   }
-
+  
   if ([FBConfiguration shouldUseTestManagerForVisibilityDetection]) {
     return [(NSNumber *)[self fb_attributeValue:FB_XCAXAIsVisibleAttribute] boolValue];
   }
-
-  XCElementSnapshot *parentWindow = [self fb_parentMatchingType:XCUIElementTypeWindow];
-  if (nil != parentWindow &&
-      CGRectIsEmpty([self fb_frameInContainer:parentWindow hierarchyIntersection:nil])) {
-    return NO;
-  }
   
+  NSMutableArray<XCElementSnapshot *> *parentsChain = [NSMutableArray array];
+  XCElementSnapshot *parent = self.parent;
+  while (parent) {
+    [parentsChain addObject:parent];
+    parent = parent.parent;
+  }
+  XCElementSnapshot *parentWindow = nil;
+  for (parent in parentsChain) {
```

Comment:
I might be reading this wrong, but I am confused by these two loops. The first would seem to create an array of parents, starting with the element and going to the top level: `[element, direct parent, ..., top level parent]`.

Then the second loop goes through that array, presumably getting `element` first, then `direct parent`, etc. It checks if the type is `XCUIElementTypeWindow` and, if so, makes note of the element and quits the loop.

Wouldn't it do the same thing to have a single loop that goes through and checks for the first ancestor that is a `XCUIElementTypeWindow`?
```objc
XCElementSnapshot *parentWindow = nil;
while (parent) {
    if (parent.elementType == XCUIElementTypeWindow) {
        parentWindow = parent;
        break;
    }
    parent = parent.parent;
}
```

Reply:
we need that array later for detecting the parent cell, but yes, it's possible to do the change